homeassistant/components/dlna_dmr/config_flow.py

```diff
@@ -0,0 +1,340 @@
+        with them here.
+        """
+        LOGGER.debug("async_step_user: user_input: %s", user_input)
+
+        errors = {}
+        if user_input is not None:
+            try:
+                discovery = await self._async_connect(user_input[CONF_URL])
+            except ConnectError as err:
+                errors["base"] = err.args[0]
+            else:
+                # If unmigrated config was imported earlier then use it
+                import_data = get_domain_data(self.hass).unmigrated_config.get(
+                    user_input[CONF_URL]
+                )
+                if import_data is not None:
+                    return await self.async_step_import(import_data)
+                # Device setup manually, assume we don't get SSDP broadcast notifications
+                options = {CONF_POLL_AVAILABILITY: True}
+                return await self._async_create_entry_from_discovery(discovery, options)
+
+        data_schema = vol.Schema({CONF_URL: str})
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_import(self, import_data: FlowInput = None) -> FlowResult:
+        """Import a new DLNA DMR device from a config entry.
+
+        This flow is triggered by `async_setup`. If no device has been
+        configured before, find any matching device and create a config_entry
+        for it. Otherwise, do nothing.
+        """
+        LOGGER.debug("async_step_import: import_data: %s", import_data)
+
+        if not import_data or CONF_URL not in import_data:
+            LOGGER.debug("Entry not imported: incomplete_config")
+            return self.async_abort(reason="incomplete_config")
+
+        self._async_abort_entries_match({CONF_URL: import_data[CONF_URL]})
+
+        location = import_data[CONF_URL]
+        self._discoveries = await self._async_get_discoveries()
+
+        poll_availability = True
+
+        # Find the device in the list of unconfigured devices
+        for discovery in self._discoveries:
+            if discovery[ssdp.ATTR_SSDP_LOCATION] == location:
+                # Device found via SSDP, it shouldn't need polling
+                poll_availability = False
+                LOGGER.debug(
+                    "Entry %s found via SSDP, with UDN %s",
+                    import_data[CONF_URL],
+                    discovery[ssdp.ATTR_SSDP_UDN],
+                )
+                break
+        else:
+            # Not in discoveries. Try connecting directly.
+            try:
+                discovery = await self._async_connect(location)
+            except ConnectError as err:
+                LOGGER.debug(
+                    "Entry %s not imported: %s", import_data[CONF_URL], err.args[0]
+                )
+                # Store the config to apply if the device is added later
+                get_domain_data(self.hass).unmigrated_config[location] = import_data
```

Comment:
This is not a good pattern. We want the config flow to be isolated from the rest of the integration. The flow should just return the config flow result.